File: sprint01/JMS/PGS_43165.cpp
#include <string>
#include <vector>

using namespace std;

int cnt = 0;

void RecurSearch(vector<int> numbers, int target, int index, int sum) {
    if (index > numbers.size() - 1)
        return;

    if (index == numbers.size() - 1 && (sum + numbers[index] == target || sum - numbers[index] == target))
        cnt++;

    RecurSearch(numbers, target, index + 1, sum + numbers[index]);
    RecurSearch(numbers, target, index + 1, sum - numbers[index]);
}

int solution(vector<int> numbers, int target) {
    int answer = 0;

    RecurSearch(numbers, target, 0, 0);
    answer = cnt;

    return answer;
}

File: sprint01/KKW/PGS_43165.py
def targetNumber(numbers, target,i, val):
    cnt=0
    if len(numbers)==i:
        if val==target:
            return cnt+1
        return cnt
    cnt+=targetNumber(numbers, target, i + 1, val + numbers[i])
    cnt+=targetNumber(numbers, target, i + 1, val - numbers[i])

    return cnt

def solution(numbers, target):
    return targetNumber(numbers,target,0,0)

print(solution([4,1,2,1],4))